core/consensus/synchronizer/impl/synchronizer_impl.cpp

```diff
@@ -69,7 +69,7 @@ namespace kagome::consensus {
     }
   }
 
-  void SynchronizerImpl::requestBlocks(const libp2p::peer::PeerInfo &peer,
+  void SynchronizerImpl::requestBlocks(const libp2p::peer::PeerId &peer,
```

Comment:
Do not use libp2p peer id for identification. To send message to a specific peer, use reference to a PeetClient object

Reply:
Synchronizer has the network state which matches peer id to peer client. Why you don't like this approach? IMO operating with PeerId is more convenient. Thus, we can avoid exposing PeerClient to the user of synchronizer. 
Also we provide network state in constructor, and this network state should provide access to peer clients. But you suggest to use PeerClient in requestBlocks method. Then we should rework idea about network state as it would look like it is not needed.